src/rust/engine/options/src/build_root.rs

```diff
@@ -14,8 +14,13 @@ impl BuildRoot {
   const SENTINEL_FILES: &'static [&'static str] = &["pants.toml", "BUILDROOT", "BUILD_ROOT"];
 
   pub fn find() -> Result<BuildRoot, String> {
-    let cwd = env::current_dir().map_err(|e| format!("Failed to determine $CWD: {e}"))?;
-    Self::find_from(&cwd)
+    match env::var("PANTS_BUILDROOT_OVERRIDE") {
```

Comment:
This function attempts to parse the environment variable value as a `String`, which requires that it is UTF-8, and this isn't guaranteed (e.g. someone could hypothetically have their repo file path in a non-UTF8 encoding). If it's not UTF-8, this returns an `Err`, and that means the value is implicitly ignored, falling back to the `current_dir` case.

I think this could be fixed using https://doc.rust-lang.org/stable/std/env/fn.var_os.html, which returns a raw `OsString`:

```suggestion
    match env::var_os("PANTS_BUILDROOT_OVERRIDE") {
```


(There's broader problems with non-UTF-8 env vars in Pants #19199, but we can at least avoid adding more problems here.)